### PR TITLE
Button render() no longer replaces original element (Connects #711)

### PR DESCRIPTION
**Pull request by slj**
Supports #711 

Replacing the initial element with loading markup has been unreliable in CI testing.  Hiding the initial element instead and appending the loading markup has consistently passed automated tests.

### CODE REVIEW

File: client/app/components/Button.jsx [hunk 1]
```diff
@@ -22,19 +33,34 @@ export default class Button extends React.Component {
...
       </span>;
-    }
+    };
 
     if (!children) {
       children = name;
     }
 
-    if (disabled) {
+    if (disabled || loading) {
```
Comment (shanear) [inline client/app/components/Button.jsx]: we shouldn't need `|| loading` here

File: client/test/components/Button-test.js [hunk 2]
```diff
@@ -13,9 +13,7 @@ describe('Button', () => {
                               onChange={onChange}
                               loading={true} />);
 
-    expect(wrapper.find('.usa-button-primary')).to.have.length(0);
-    expect(wrapper.find('#test-button')).to.have.length(0);
-    expect(wrapper.find('.cf-react-loading-indicator')).to.have.length(1);
+    expect(wrapper.find('.hidden-field')).to.have.length(1);
```
Comment (shanear) [inline client/test/components/Button-test.js]: can we get more specific with this test? like:

`wrapper.find(#test-button'.hidden-field')`

File: client/app/components/Button.jsx [hunk 3]
```diff
@@ -22,19 +33,34 @@ export default class Button extends React.Component {
       type
     } = this.props;
 
-    if (loading) {
-      classNames = classNames.filter((className) => !className.includes('usa-button'));
+    let LoadingIndicator = () => {
+      if (!app) {
```
Comment (shanear) [inline client/app/components/Button.jsx]: `app ||= 'dispatch'` is a more terse, idiomatic way of doing this (lines 36-39)

File: client/app/components/Button.jsx [hunk 4]
```diff
@@ -10,8 +9,20 @@ export default class Button extends React.Component {
     }
   }
 
+  loadingClasses = (app, loading) => {
+    let classes = ` cf-${app}`;
+
+    if (loading) {
```
Comment (shanear) [inline client/app/components/Button.jsx]: we shouldn't need this check here. By the time you call `loadingClasses` (via line 79 -> line 45), we've already validated that loading is true.

Once you remove this check, we sort of don't really need this method anymore. We can incorporate it directly into `LoadingIndicator`

File: client/app/components/Button.jsx (hunk 1 above)
Comment (slj) [inline client/app/components/Button.jsx]: It's needed in order to apply the `usa-button-disabled` class

File: client/app/components/Button.jsx (hunk 3 above)
Comment (slj) [inline client/app/components/Button.jsx]: That's a ruby idiom, not JS, right?

File: client/test/components/Button-test.js (hunk 2 above)
Comment (slj) [inline client/test/components/Button-test.js]: It's a shallow test so only the topmost element is created; I think testing for the key class suffices.

File: client/app/components/Button.jsx (hunk 4 above)
Comment (slj) [inline client/app/components/Button.jsx]: This was careless of me.  Good catch.  Thx.

File: client/app/components/Button.jsx (hunk 3 above)
Comment (shanear) [inline client/app/components/Button.jsx]: whoops, good call. I mean:
`app = app || 'dispatch'`

File: client/app/components/Button.jsx (hunk 1 above)
Comment (shanear) [inline client/app/components/Button.jsx]: Good point. sounds good.

File: client/test/components/Button-test.js (hunk 2 above)
Comment (shanear) [inline client/test/components/Button-test.js]: it's not that it doesn't work. It's that it isn't clear what is being tested, and it's also easier to introduce a false positive, like if I added another element with .hidden-field into this element.

Comment (slj) [inline client/test/components/Button-test.js]: I'll try to change it though.

**Pull request merged**